src/json_output.rs

```diff
@@ -237,36 +247,59 @@ impl BinXmlOutput for JsonOutput {
             //    },
             //    "#text": "4902"
             //  },
-            let current_object =
-                current_value
-                    .as_object_mut()
-                    .context(err::JsonStructureError {
-                        message: "expected current value to be an object type",
-                    })?;
-
-            current_object.insert("#text".to_owned(), value.clone().into());
+            if let Some(object) = current_value.as_object_mut() {
+                object.insert("#text".to_owned(), value.clone().into());
+                return Ok(());
+            };
+
+            // If we already have a string (because we got two consecutive `Character` events,
+            // Concat them.
+            if let Some(s) = current_value.as_str() {
+                let new_string = s.to_string();
+                mem::replace(
+                    current_value,
+                    Value::String(new_string + &value.as_cow_str()),
+                )
+            } else {
+                return Err(SerializationError::JsonStructureError {
+                    message: format!(
+                        "expected current value to be an object type or a String, found {:?}, value is {:?}",
+                        current_value, value
+                    ),
+                });
+            };
         }
 
         Ok(())
     }
 
-    fn visit_cdata_section(&mut self) -> Result<()> {
-        unimplemented_fn!("visit_cdata_section")
+    fn visit_cdata_section(&mut self) -> SerializationResult<()> {
+        Err(SerializationError::Unimplemented {
+            message: format!("`{}`: visit_cdata_section", file!()),
+        })
     }
 
-    fn visit_entity_reference(&mut self) -> Result<()> {
-        unimplemented_fn!("visit_entity_reference")
+    fn visit_entity_reference(&mut self, entity: &BinXmlName) -> Result<(), SerializationError> {
+        // JSON doesn't care about entity references.
+        Ok(())
```

Comment:
You don't need to add the actual value of the reference?

Reply:
fixed